### PR TITLE
[SYCL][Devops] Fix `DockerFile` linting issues discovered by trivy

### PR DESCRIPTION
**Pull request by AlexeySachkov**
Linter/checker used: [trivy v0.58.0](https://github.com/aquasecurity/trivy/releases/tag/v0.58.0)

Issues addressed:
- AVD-DS-0017 (HIGH): The instruction 'RUN <package-manager> update' should always be followed by '<package-manager> install' in the same RUN statement.
  See https://avd.aquasec.com/misconfig/ds017
- AVD-DS-0002 (HIGH): Specify at least 1 USER command in Dockerfile with non-root user as argument
  See https://avd.aquasec.com/misconfig/ds002
- AVD-DS-0002 (HIGH): Last USER command in Dockerfile should not be 'root'

Issues remaining:
- AVD-DS-0026 (LOW): Add HEALTHCHECK instruction in your Dockerfile
  See https://avd.aquasec.com/misconfig/ds026

I didn't add `HEALTHCHECK` command to our containers, because I don't know if that makes sense and which command to launch. I.e. our containers they only provide some pre-installed tools, but they don't launch any services which we could check.

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
Note for reviewers: I have very little idea about how to validate this (though I see that some CI containers job have been launched), so any suggestions here are appreciated.

**Review by sarnex — APPROVED** [review]
i think we have to wait until merge, nothing uses the PR containers i think, only the official ones

seems sane to me

**Comment by sarnex** [comment]
CI failed in making the docker image, but for running them yeah I don't think we can do that here

**Comment by AlexeySachkov** [comment]
> CI failed in making the docker image, but for running them yeah I don't think we can do that here

Apparently some containers re-use other containers, that issue was fixed. Failure on AMD seems to be an infrastructure one (runner got disconnected). Moreover, we don't use images built as part of the PR in our pre-commit, so it shouldn't really be related.

@sarnex, @intel/dpcpp-devops-reviewers, @intel/llvm-gatekeepers, we need to agree on a strategy here. Considering that this change has a potential to disrupt all pre-commits, I suggest that we merge it ASAP and then monitor all pre-commits to quickly revert if we spot any issues.

**Comment by sarnex** [comment]
@AlexeySachkov That's fine with me, can you watch the precommit CI or should I? If you, let me know when you're ready for merge.

**Comment by sarnex** [comment]
cool, if something crazy happens feel free to ping me

**Comment by AlexeySachkov** [comment]
https://github.com/intel/llvm/actions/runs/12239030332 - post-commit job to re-build containers has finished successfully. If I understand correctly, starting from this point all new pre-commits should use new containers.

**Comment by sarnex** [comment]
yeah i think we need to wait for a new precommit/postcommit run. you can run them manually in a fake pr if you want or just wait and check

**Comment by sarnex** [comment]
@AlexeySachkov Looks like the nightly failed, can you take a look?

https://github.com/intel/llvm/actions/runs/12248716063/job/34169121051#step:4:194

**Comment by AlexeySachkov** [comment]
> @AlexeySachkov Looks like the nightly failed, can you take a look?
> 
> https://github.com/intel/llvm/actions/runs/12248716063/job/34169121051#step:4:194

Thanks for letting me know, @sarnex, we seem to have one more case of one docker image re-using another which is not covered by pre-commit. I will address

**Comment by AlexeySachkov** [comment]
> @AlexeySachkov Looks like the nightly failed, can you take a look?
> 
> https://github.com/intel/llvm/actions/runs/12248716063/job/34169121051#step:4:194

#16324 should hopefully fix that

**Pull request merged**
